### PR TITLE
SMP: Determine initiator by direction instead of role

### PR DESCRIPTION
**Pull request by zxzxwu**
Though in the spec this is not allowed, but in some ambiguous cases such as SMP over BR/EDR or we want to test how remote devices handle invalid pairing requests, we still need to allow this behavior, with some logs to let users know it's invalid.

### CODE REVIEW

**Review by uael — APPROVED** [review]
LGTM 👍

File: bumble/smp.py [hunk 1]
```diff
@@ -1680,6 +1680,8 @@ def send_command(self, connection, command):
     def on_smp_pdu(self, connection, pdu):
         # Look for a session with this connection, and create one if none exists
         if not (session := self.sessions.get(connection.handle)):
+            if connection.role == BT_CENTRAL_ROLE:
+                logger.warn('Remote starts pairing as Peripheral!')
```
Comment (barbibulle) [inline bumble/smp.py:1684]: `.warn` is deprecated, use `.warning` instead.

File: bumble/smp.py [hunk 2]
```diff
@@ -1709,10 +1711,12 @@ def ecc_key(self):
 
     async def pair(self, connection):
         # TODO: check if there's already a session for this connection
+        if connection.role != BT_CENTRAL_ROLE:
+            logger.warn('Start pairing as Peripheral!')
```
Comment (barbibulle) [inline bumble/smp.py:1715]: `.warning`

**Review by barbibulle — APPROVED** [review]

File: bumble/smp.py (hunk 2 above)
Comment (zxzxwu) [reply bumble/smp.py]: Done

**Pull request merged**